log/logfmt_logger.go

```diff
@@ -0,0 +1,29 @@
+package log
+
+import (
+	"io"
+
+	"gopkg.in/logfmt.v0"
+)
+
+type logfmtLogger struct {
+	w io.Writer
+}
+
+// NewLogfmtLogger returns a logger that encodes keyvals to the Writer in
+// logfmt format.
+func NewLogfmtLogger(w io.Writer) Logger {
+	return &logfmtLogger{w}
+}
+
+func (l logfmtLogger) Log(keyvals ...interface{}) error {
+	b, err := logfmt.MarshalKeyvals(keyvals...)
```

Comment:
The intermediate buffer is there to avoid interleaving calls to the `Writer` from concurrent calls to `Log`. The goal is to have a 1:1 connection between calls to `Log` and calls to `w.Write`.

I will add a comment to document that constraint.